src/Agent.Worker/Build/BuildJobExtension.cs

```diff
@@ -188,6 +188,7 @@ public void ConvertLocalPath(IExecutionContext context, string localPath, out st
             }
 
             executionContext.Debug($"Sync source for endpoint: {SourceEndpoint.Name}");
+            SourceProvider.SetVariablesInEndpoint(executionContext, SourceEndpoint);
```

Comment:
I now realize, I think it does matter whether the variables are left in the endpoint. I think you need to clone the endpoint and set the variables in the cloned copy.

A not so distant goal for build is sync'ing multiple repos, and for release syncing multiple source artifacts of the same type (e.g. two TFVC source artifacts). In this future work, ***if two repo's (for build) or two artifacts (for release) share the same endpoint*** (we don't know yet since it isn't implemented), then overriding the non-cloned endpoint data likely won't be the solution.

My concern is that we start jamming the values into the non-cloned endpoint data today, then pull back on that later, and break task authors who took a dependency on the values in the endpoint data.